starter-code/basic-algorithms.js

```diff
@@ -1,7 +1,88 @@
 // Names and Input
+var hacker1 = "John"
+console.log("The driver's name is " + hacker1);
 
+//var hacker2 = prompt("What is your name?")
+var hacker2 = "Luca"
+console.log("The driver's name is " + hacker2);
 
-//Conditionals
 
 
+//Conditional
+if (hacker1.length > hacker2.length){
+  console.log("The Driver has the longest name, it has " + hacker1.length + " characters")
+} else if (hacker2.length > hacker1.length){
+  console.log("Yo, navigator got the longest name, it has " + hacker2.length + " characters")
+} else { console.log("wow, you both got equally long names, " + hacker1.length + " characters!!") }
+
+
+// Loops
+var hacker1 = "John"
+
+for (var i = 0; i < hacker1.length; i++){
+  console.log(hacker1[i].toUpperCase() + " ")
+}
+
+var hacker1 = "John"
+
+for (var i = hacker1.length - 1; i >= 0; i--){
+  console.log(hacker1[i])
+} // output: nhoJ
+
+
+if (hacker1>hacker2) {
+  console.log("The driver's name goes first");
+} else if (hacker2>hacker1) {
+  console.log("Yo, the navigator goes first definitely")
+} else {
+    console.log("What?! You both got the same name?")
+}
+
+function isPalindrome(str){
+
+  // lower the case
+  var string = str.toLowerCase()
```

Comment:
As a good practice, it is better not to call a variable my the name of a JS type name, you can use for intance "myString" instead of "string" :)